include/aws/common/common.h

```diff
@@ -138,10 +138,10 @@ AWS_STATIC_ASSERT(sizeof(char) == 1);
 
 /* Allocator structure. An instance of this will be passed around for anything needing memory allocation */
 struct aws_allocator {
-    void *(*mem_acquire)(struct aws_allocator *allocator, size_t size);
+    void *(*mem_acquire)(struct aws_allocator *allocator, size_t size, const char *file, int line);
```

Comment:
My suggestion here would be to have a structure which encapsulates file/line, and potentially future things like name/category/flags or any other distinguishing info we may come up with. You can construct the structure and pass it by value via macro. That way the API doesn't need to change should we need to collect more info, we could just add an additional macro or something to provide it.

Reply:
Hey good idea :)